tfhe/src/core_crypto/fft_128_split_impl/crypto/bootstrap.rs

```diff
@@ -0,0 +1,236 @@
+use super::super::math::fft::Fft128View;
+use super::ggsw::cmux_split;
+use crate::core_crypto::algorithms::extract_lwe_sample_from_glwe_ciphertext;
+use crate::core_crypto::commons::parameters::{LutCountLog, ModulusSwitchOffset, MonomialDegree};
+use crate::core_crypto::commons::traits::ContiguousEntityContainerMut;
+use crate::core_crypto::commons::utils::izip;
+use crate::core_crypto::entities::*;
+use crate::core_crypto::fft_128_impl::crypto::bootstrap::Fourier128LweBootstrapKey;
+use crate::core_crypto::fft_128_split_impl::math::fft::wrapping_neg;
+use crate::core_crypto::fft_impl::crypto::bootstrap::pbs_modulus_switch;
+use crate::core_crypto::prelude::{Container, ContainerMut};
+use aligned_vec::CACHELINE_ALIGN;
+use dyn_stack::{PodStack, ReborrowMut};
+
+pub fn polynomial_wrapping_monic_monomial_mul_assign_split(
+    output_lo: Polynomial<&mut [u64]>,
+    output_hi: Polynomial<&mut [u64]>,
+    monomial_degree: MonomialDegree,
+) {
+    let output_lo = output_lo.into_container();
+    let output_hi = output_hi.into_container();
+    let full_cycles_count = monomial_degree.0 / output_lo.container_len();
+    if full_cycles_count % 2 != 0 {
+        izip!(&mut *output_lo, &mut *output_hi)
+            .for_each(|(lo, hi)| (*lo, *hi) = wrapping_neg((*lo, *hi)));
+    }
+    let remaining_degree = monomial_degree.0 % output_lo.container_len();
+    output_lo.rotate_right(remaining_degree);
+    output_hi.rotate_right(remaining_degree);
+    izip!(output_lo, output_hi)
+        .take(remaining_degree)
+        .for_each(|(lo, hi)| (*lo, *hi) = wrapping_neg((*lo, *hi)));
+}
+
+pub fn polynomial_wrapping_monic_monomial_div_assign_split(
+    output_lo: Polynomial<&mut [u64]>,
+    output_hi: Polynomial<&mut [u64]>,
+    monomial_degree: MonomialDegree,
+) {
+    let output_lo = output_lo.into_container();
+    let output_hi = output_hi.into_container();
+    let full_cycles_count = monomial_degree.0 / output_lo.container_len();
+    if full_cycles_count % 2 != 0 {
+        izip!(&mut *output_lo, &mut *output_hi)
+            .for_each(|(lo, hi)| (*lo, *hi) = wrapping_neg((*lo, *hi)));
+    }
+    let remaining_degree = monomial_degree.0 % output_lo.container_len();
+    output_lo.rotate_left(remaining_degree);
+    output_hi.rotate_left(remaining_degree);
+    izip!(output_lo, output_hi)
+        .rev()
+        .take(remaining_degree)
+        .for_each(|(lo, hi)| (*lo, *hi) = wrapping_neg((*lo, *hi)));
+}
```

Comment:
would it make sense to provide these in the polynomial_algorithms module ?

Reply:
it depends on if we're gonna need them. what are the original polynomial mul/div functions used for?

Reply:
The wrapping monic ones are mainly for the PBS IIRC

Reply:
so those are probably too specific to make them shared right ?